### PR TITLE
Show Chart in Modal for Report Link Widgets

### PR DESCRIPTION
**Pull request by shubhamsc**

### CODE REVIEW

**Comment by HerbertYiga** [comment]
kindly look at this to make your pull requests better https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: src/report-links/report-links.css [hunk 1]
```diff
@@ -21,25 +20,22 @@
...
+.report-link-container {
+  width: 100%;
+  height: 100%;
+  display: flex;
 }
 
-.disabled {
-  color: grey !important;
-  cursor: default !important;
+button.closeModal {
```
Comment (rrameshbtech) [inline src/report-links/report-links.css]: Lets use same naming convention for all css classes

File: src/report-links/report-links.test.tsx [hunk 2]
```diff
@@ -56,192 +88,54 @@ describe("Report Links", () => {
...
-      {
-        name: "First Test Report",
-        uuid: "report-uuid"
-      }
-    ];
-    mockEsmAPI.openmrsFetch.mockResolvedValueOnce({
-      data: { uuid: "report-request-uuid" }
-    });
-
+  it("should not render modal when report link is not clicked", () => {
```
Comment (rrameshbtech) [inline src/report-links/report-links.test.tsx]: can it be named like 
```suggestion
  it("should not show chart modal during initial load", () => {
```

File: src/report-links/report-links.test.tsx [hunk 3]
```diff
@@ -56,192 +88,54 @@ describe("Report Links", () => {
...
 
-  it("should open report results in tab when view report clicked after report rendered", async () => {
-    const mockReports = [
-      {
-        name: "First Test Report",
-        uuid: "report-uuid"
-      }
-    ];
-
+  it("should show modal pop-up when report link is clicked", async () => {
```
Comment (rrameshbtech) [inline src/report-links/report-links.test.tsx]:
```suggestion
  it("should show modal with chart when report link is clicked", async () => {
```

File: src/report-links/report-links.tsx [hunk 4]
```diff
@@ -1,181 +1,96 @@
...
 import WidgetHeader from "../commons/widget-header/widget-header.component";
 import resources from "./translations";
 import { initI18n } from "../utils/translations";
+import ChartLoader from "../charts/chart-loader.component";
 
 import globalStyles from "../global.css";
 import styles from "./report-links.css";
+import ReactModal from "react-modal";
+
+ReactModal.defaultStyles.overlay.backgroundColor = "#ffffff00";
```
Comment (rrameshbtech) [inline src/report-links/report-links.tsx]: Was the colour value was `#ffffff`?
```suggestion
ReactModal.defaultStyles.overlay.backgroundColor = "#ffffff";
```

File: src/report-links/report-links.tsx [hunk 5]
```diff
@@ -1,181 +1,96 @@
...
+const customStyles = {
```
Comment (rrameshbtech) [inline src/report-links/report-links.tsx]: is it modal's custom styles? if Yes, can this be named like
```suggestion
const modalStyles = {
```
Else we can prefer getting the styles from css class if possible.

File: src/report-links/report-links.tsx [hunk 6]
```diff
@@ -1,181 +1,96 @@
...
+            config={currentChart}
+            locale={locale}
+          ></ChartLoader>
+        )}
         <button
-          title="Report not rendered yet"
-          className={styles["disabled"]}
-          onClick={showNoReportAvailableError}
+          className={`${styles["closeModal"]}`}
+          onClick={() => setIsModalOpen(false)}
```
Comment (rrameshbtech) [inline src/report-links/report-links.tsx]: Please add title for any button added.

File: src/report-links/report-links.tsx [hunk 7]
```diff
@@ -1,181 +1,96 @@
...
-  const reportLinkElement = (reportLink: ReportLink) => (
-    <div className={styles["report-link"]} key={getKey(reportLink.name)}>
+  const reportLinkElement = chartConfig => (
+    <div className={styles["report-link"]} key={getKey(chartConfig.name)}>
       <span className={styles["report-name"]}>
         <i className={"icon-link"}></i>
-        <span>
-          <Trans>{reportLink.name}</Trans>
-        </span>
+        <button onClick={() => openChartInModal(chartConfig)}>
```
Comment (rrameshbtech) [inline src/report-links/report-links.tsx]: set title for the button.

**Pull request merged**